README.md

```diff
@@ -25,16 +25,38 @@ string result = converter.Convert(html);
 bool githubFlavored = true; // generate GitHub flasvoured markdown, supported for BR, PRE and table tags
 bool removeComments = true; // will ignore all comments
 var config = new ReverseMarkdown.Config(UnknownTagsOption.PassThrough, 
-                githubFlavoured:githubFlavoured, removeComments:removeComments);
+                githubFlavoured:githubFlavoured, removeComments:removeComments) {
+                    new Config() {
+                        HrefHandling = Config.HrefHandlingOption.Smart,
+                        WhitelistUriSchemes = new string[] {"http", "https", "ftp", "ftps", "file"}
+                    }
+
+                };
 var converter = new ReverseMarkdown.Converter(config);
 ```
 
-### UnknownTags config - how to handle unknown tags. 
-Valid options are:
-* `UnknownTagsOption.PassThrough` - Include the unknown tag completely into the result. This is the default
-* `UnknownTagsOption.Drop` - Drop the unknown tag and its content
-* `UnknownTagsOption.Bypass` - Ignore the unknown tag but try to convert its content
-* `UnknownTagsOption.Raise` - Raise an error to let you know
+## Configuration options
+* `GithubFlavored` - Github style markdown for br, pre and table. Default is false
+* `RemoveComments` - Remove comment tags with text. Default is true
+* `BarePlaintext` - Convert to bare plaintext, not markdown. Default is false
+* `CompressNewlines` - Cleans output so that it doesn't contain double newline characters. Default is true
+* `HrefHandling` - how to handle `<a>` tag href attribute
+  * `None` - Outputs `[{name}]({href}{title})` even if name and href is identical. This is the default option.
```

Comment:
My reasoning for having Enumerations:

1. They allow adding options in the future in case of different `href` behavior is implemented.
2. Code is more readable. I thought if I add another boolean and within the constructor, it gets ugly when, _firstly_, there are so much parameters in constructor and _secondly_, true, false, true, false doesn't kind of indicate what it means. Ofcourse C# provides named parameters which can be used by library consumers to make their code more clear, as in your example.

Ofcourse I will make it the way you like. Would you prefer having "SmartHrefHandling" boolean setting within the existing constructor?

Reply:
As a next step, I would like to expose all config properties as public settable so that a specific constructor bound params may not be required.

With regards to smart href handling, I don't foresee any other options on this so we can just use your suggestion of of name `SmartHrefHandling`.